optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -100,6 +100,11 @@
     public EntityDescriptor(SolutionDescriptor<Solution_> solutionDescriptor, Class<?> entityClass) {
         this.solutionDescriptor = solutionDescriptor;
         this.entityClass = entityClass;
+        isInitializedPredicate = this::isInitialized;
+    }
+
+    public Predicate<Object> getIsInitializedPredicate() {
+        return isInitializedPredicate;
```

Comment:
Yes, this is strange. It's due to node sharing: doing it the more obvious way creates 2 different lamdba instances.

I'll add a comment to point that out, good catch!

Reply:
```
    /**
     * Using entityDescriptor::isInitialized directly breaks node sharing
     * because it creates multiple instances of this {@link Predicate}.
     * @return never null, always the same {@link Predicate} instance to {@link #isInitialized(Object)}
     */
    public Predicate<Object> getIsInitializedPredicate() {
        return isInitializedPredicate;
    }
```